engine.js

```diff
@@ -0,0 +1,96 @@
+"format cjs";
+
+var wrap = require('word-wrap');
+var cloneDeep = require('lodash.clonedeep');
+var map = require('lodash.map');
+var longest = require('longest');
+var rightPad = require('right-pad');
+
+// This can be any kind of SystemJS compatible module.
+// We use Commonjs here, but ES6 or AMD would do just
+// fine.
+module.exports = function (options) {
+
+  options = cloneDeep(options || {});
+  var types = options.types;
+
+  var length = longest(Object.keys(types)).length + 1;
+  var choices = map(types, function (type, key) {
```

Comment:
Intuition 😆 nah, I don't really know, it just felt more appropriate for a collection to be an array. I think that `order` keys are quite ugly since the data-structure itself doesn't impost any integrity on them (e.g. you can have two items with `order: 3`).

Let's go with it as it is for now though :)